lesson_13/edmond_sabou/home1_les13.py

```diff
@@ -0,0 +1,64 @@
+from math import sqrt
+
+
+class Polygon:
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return f"{no_of_sides} is the number of sides.\n"
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print(f"Side {side_index} with length: {length}.\n")
+
+
+class Square(Polygon):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        class_name = type(self).__name__
+        side, *_ = self.sides
+        return side ** 2
+
+    @classmethod
+    def from_area(cls, sq):
+        sq = sqrt(sq)
+        return sq
```

Comment:
Your class method returns `sq`.  In this case `sq` is a number, not an instance of Square. To have an alternative constructor you need to return an instance of the class. You have to options cls(sq,sq,sq,sq) or Square(sq,sq,sq,sq)